snapshot_dbg_cli/breakpoint_utils.py

```diff
@@ -116,17 +118,27 @@ def normalize_breakpoint(bp, bpid=None):
   If a breakpoint is returned, the following fields are guaranteed to
   be populated:
 
-  id
-  location
-    path
-    line
-  action
-  isFinalState
-  createTime
-  createTimeUnixMsec
-  finalTime
-  finalTimeUnixMsec
-  userEmail
+    id
+    location
+      path
+      line
+    action
+    isFinalState
+    createTime
+    createTimeUnixMsec
+    finalTime
+    finalTimeUnixMsec
+    userEmail
+
+  If the 'action' field is 'CAPTURE', the breakpoint represents a snapshot, and
+  if it's 'LOG', the breakpoint represents a logpoint.
+
+  Additionally, if the breakpoint represents a logpoint (action is 'LOG'), the
+  following fields will be present:
+
+    logMessageFormat        - Form of 'a: $0, b: $1', used by the agents
```

Comment:
Worth mentioning expressions?

Reply:
The comments here were meant to specify what guarantees normalize_breakpoint makes about what fields are present if it succeeds. Eg. Code that does need to parse expressions will know that it is optional and proceed accordingly, and so it's not mentioned here. I think what we do need though is a documented 'spec' of what fields a breakpoint can have...